exe/task_wrapper.cc

```diff
@@ -0,0 +1,47 @@
+#include <leatherman/json_container/json_container.hpp>
+#include <leatherman/execution/execution.hpp>
+#include <leatherman/file_util/file.hpp>
+#include <leatherman/locale/locale.hpp>
+
+#include <boost/nowide/iostream.hpp>
+#include <boost/nowide/fstream.hpp>
+
+namespace lth_loc = leatherman::locale;
+namespace lth_jc = leatherman::json_container;
+namespace lth_exec = leatherman::execution;
+namespace lth_file = leatherman::file_util;
+namespace lth_util = leatherman::util;
+
+int main(int argc, char *argv[])
+{
+    boost::nowide::cin >> std::noskipws;
+    std::istream_iterator<char> i_s_i(boost::nowide::cin), end;
+    auto params = lth_jc::JsonContainer(std::string { i_s_i, end });
+    auto task_executable = params.get<std::string>("executable");
+    int exitcode;
+
+    try {
+        auto exec = lth_exec::execute(
+            task_executable,
+            params.get<std::vector<std::string>>("arguments"),
+            params.get<std::string>("input"),
+            params.get<std::string>("stdout"),
+            params.get<std::string>("stderr"),
+            {},       // environment
+            nullptr,  // PID callback
+            0,        // timeout
+            lth_util::option_set<lth_exec::execution_options> {
+                lth_exec::execution_options::thread_safe,
+                lth_exec::execution_options::merge_environment,
+                lth_exec::execution_options::inherit_locale });
+        exitcode = exec.exit_code;
+    } catch (lth_exec::execution_exception &e) {
+        lth_file::atomic_write_to_file(lth_loc::format("Task '{1}' failed to run: {2}", task_executable, e.what()),
+                                       params.get<std::string>("stderr"));
```

Comment:
We don't need to write atomically to the `stderr` file, so I'd rather not do that. It gave me some grief when I was testing the `task_wrapper` like this:
```sh
echo '{ "executable": "/bin/shhhh", "arguments": [ "-c", "echo hi; exit 5"], "input": "", "stdout": "/dev/stdout", "stderr": "/dev/stderr", "exitcode": "ec" }' | bin/task_wrapper
```
Obviously it's not a good idea to replace `/dev/stderr`.